src/scripts/graph.ts

```
@@ -272,25 +273,42 @@ function updatePath(path: HTMLDependencyElement, dest?: Point) {
   }
 }
 
-const panzoom = {
-  pan: { x: 0, y: 0 },
-  zoom: 1,
-};
+const getPanzoom = () => {
+  const graph = getElementById("graph");
+  const panzoomX = graph.dataset.panzoomX;
+  const panzoomY = graph.dataset.panzoomY;
+  const panzoomZoom = graph.dataset.panzoomZoom;
 
-function updatePanzoom() {
-  const itemsContainer = getElementById("itemsContainer");
-  itemsContainer.style.transform = `translate(${panzoom.pan.x}px, ${panzoom.pan.y}px) scale(${panzoom.zoom})`;
-}
+  if (
+    panzoomX === undefined ||
+    panzoomY === undefined ||
+    panzoomZoom === undefined
+  )
+    throw new Error("panzoom values missing");
```

thought: then what happens?

---

In theory never, but it makes typescript happy

The alternative would be to use the ! operator on the values but it's the same it will throw an exception on undefined

---

Oh sorry I misread the question:

If this error is thrown the error is not handled and bubbles up to the console

But that's unlikely to ever happen and this is throwaway code anyway